gpjax/gps.py

```diff
@@ -247,6 +254,89 @@ def predict_fn(
 
         return predict_fn
 
+    def sample_approx(
+        self,
+        num_samples: int,
+        params: Dict,
+        seed: KeyArray,
+        num_features: Optional[int] = 100,
+    ) -> FunctionalSample:
+        """Build an approximate sample from the Gaussian process prior. This method
+        provides a function that returns the evaluations of a sample across any given
+        inputs.
+
+        In particular, we approximate the Gaussian processes' prior as the finite feature
+        approximation
+
+        .. math:: \hat{f}(x) = \sum_{i=1}^m \phi_i(x)\theta_i
+
+
+        where :math:`\phi_i` are m features sampled from the Fourier feature decomposition of
+        the model's kernel and :math:`\theta_i` are samples from a unit Gaussian.
+
+
+        A key property of such functional samples is that the same sample draw is
+        evaluated for all queries. Consistency is a property that is prohibitively costly
+        to ensure when sampling exactly from the GP prior, as the cost of exact sampling
+        scales cubically with the size of the sample. In contrast, finite feature representations
+        can be evaluated with constant cost regardless of the required number of queries.
+
+        In the following example, we build 10 such samples
+        and then evaluate them over the interval :math:`[0, 1]`:
+
+        Example:
+            >>> import gpjax as gpx
+            >>> import jax.numpy as jnp
+            >>>
+            >>> kernel = gpx.kernels.RBF()
+            >>> prior = gpx.Prior(kernel = kernel)
+            >>> seed = jr.PRNGKey(123)
+            >>>
+            >>> parameter_state = gpx.initialise(prior)
+            >>> sample_fn = prior.sample_appox(10, parameter_state.params, seed)
+            >>> sample_fn(jnp.linspace(0, 1, 100))
+
+        Args:
+            num_samples (int): The desired number of samples.
+            params (Dict): The specific set of parameters for which the sample
+            should be generated for.
+            seed (KeyArray): The random seed used for the sample(s).
+            num_features (int): The number of features used when approximating the
+            kernel.
+
+
+        Returns:
+            FunctionalSample: A function representing an approximate sample from the Gaussian
+            process prior.
+        """
+        for integer_input in [num_features, num_samples]:
+            if (not isinstance(integer_input, int)) or integer_input < 0:
+                raise ValueError
```

Comment:
Can you be explicit here e.g., `ValueError('input is negative and must be positive')` or `ValueError(f'input is of type {type(integer_input} and must be an integer')`?